### PR TITLE
Multiple fixes and addings to support AFL running on Android devices

### PR DESCRIPTION
**Pull request by JoeyJiao**

### CODE REVIEW

File: afl-fuzz.c [hunk 1]
```diff
@@ -7071,6 +7074,7 @@ static void check_term_size(void) {
 
   if (ioctl(1, TIOCGWINSZ, &ws)) return;
 
+  if (ws.ws_row == 0 && ws.ws_col == 0) return;
```
Comment (jonathanmetzman) [inline afl-fuzz.c]: What's the purpose of this change?

File: afl-analyze.c [hunk 2]
```diff
@@ -28,6 +28,9 @@
 */
 
 #define AFL_MAIN
+#ifdef __ANDROID__
```
Comment (jonathanmetzman) [inline afl-analyze.c]: Instead of doing
```
#ifdef __ANDROID__
  #include "android-ashmem.h"
#endif
```
where it might be needed, how about just including it everywhere and then putting the entire android-ashmem.h file in:
```
#ifdef __ANDROID__
  ...
#endif
```

**Comment by jonathanmetzman** [comment]
Thanks for the patch.
I'll do a more thorough review in the next couple days.

File: afl-fuzz.c (hunk 1 above)
Comment (JoeyJiao) [reply afl-fuzz.c]: In some version of tty like putty, when I run afl-fuzz, the status screen won't show no matter how large the window is , after check, I found that ws.ws_row and ws_ws_col always returns 0.

File: afl-analyze.c (hunk 2 above)
Comment (JoeyJiao) [reply afl-analyze.c]: Can you explain more, I didn't get the point.

Comment (jonathanmetzman) [reply afl-analyze.c]: I think it is ugly to do
```
#ifdef __ANDROID__
  #include "android-ashmem.h"
#endif 
```
in every file that mst include "android-ashmem.h". So instead just do 
```
#include "android-ashmem.h"
```
In these files.

And then in android-ashmem.h, put this around the entire file:
```
#ifdef __ANDROID__
  ...
#endif 
```
So that the header file is empty on non-android but can be included anywhere.
Here's an example of this from libFuzzer:
Ordinarily [this file](https://cs.chromium.org/chromium/src/third_party/libFuzzer/src/FuzzerExtFunctionsWindows.cpp?l=13) can't compile on non-Windows, but it's OK to build it because the entire file is wrapped in
```
#ifdef LIBFUZZER_WINDOWS
...
#endif 
```

Does this make more sense? If not I can try to explain again, or I can fix this myself after we land this code.

File: afl-fuzz.c (hunk 1 above)
Comment (jonathanmetzman) [reply afl-fuzz.c]: ACK

File: afl-as.c [hunk 3]
```diff
@@ -74,7 +74,7 @@ static u32  inst_ratio = 100,   /* Instrumentation probability (%)      */
    instrumentation for whichever mode we were compiled with. This is not
    perfect, but should do the trick for almost all use cases. */
 
-#ifdef __x86_64__
+#if (defined (__x86_64__) || defined (__arm64__) || defined (__aarch64__))
```
[the remark this answers is not in the document]
Comment (JoeyJiao) [reply afl-as.c]: Done

File: afl-whatsup [hunk 4]
```diff
@@ -54,7 +54,7 @@ fi
 
 CUR_TIME=`date +%s`
 
-TMP=`mktemp -t .afl-whatsup-XXXXXXXX` || exit 1
+TMP=`mktemp -t .afl-whatsup-XXXXXXXX` || TMP=`mktemp -p /data/local/tmp .afl-whatsup-XXXXXXXX` || exit 1
```
[the remark this answers is not in the document]
Comment (JoeyJiao) [reply afl-whatsup]: Android doesn't have /tmp writable to user while mktem -t will create temp directory under /tmp;
The right part is to create dir under /data/local/tmp which is writable to normal user.
